[Woo POS] Animate bottom tab on switch modes

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #13904

## Description
This PR addresses the design feedback on https://github.com/woocommerce/woocommerce-ios/issues/13859 regarding the bottom TabBar appearing and being hidden immediately when we switch modes. We update this behaviour with a 0.5 seconds animation

The changes required a bit of trial and error, since: 
1. When the TabBar is hidden, UIKit doesn't respond to changes in visual properties to optimize for unnecessary rendering, so we need to make the TabBar visible and it's alpha set to 0 (still "invisible") in order to apply animations, then animate it from 0 to 1.
2. Setting the insets outside of the animation's completion handler caused the bug at https://github.com/woocommerce/woocommerce-ios/pull/13785 to re-appear.

I've tested this on different simulators, my physical device, and specially the device where we could reproduce the FloatingButton bug consistently: iPad Air 11 with iOS 17.5, and I haven't seen the regression.

## Testing
- Go to Menu > Tap the Point of Sale button
- Observe the bottom TabBar disappearing  with an animation while the POS loading screen starts
- Observe that there's no regression on the location of the `FloatingView`, it appears at the bottom to the screen and it doesn't reallocate itself on attempting to switch portrait/landscape or tapping the `...` option at the top.
- Exit POS, after confirming, observe how the TabBar reappears with an animation

https://github.com/user-attachments/assets/0d247ebc-f887-47f1-8111-9c2cb2b07446

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Adds animation to tabbar when move from POS to app

iamgabrielma committed:
Adds animation to hide TabBar when entering POS

This required a bit of trial and error: Depending on how we call the animation and its completion handler, there’s a risk of a regression to gh-13785 where the UI is not being updated appropiately, leaving additional un-used space at the bottom of the screen. Please check the comment in the function for more context.

staskus commented on WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift:
```diff
@@ -243,11 +241,21 @@ private extension HubMenuViewModel {
+            UIView.animate(withDuration: 0.5) {
+                mainTabBarController.tabBar.alpha = 0
+            } completion: { _ in
+                mainTabBarController.tabBar.isHidden = isPointOfSaleActive
```
Since we hide the tab bar only after the animation completes, the loading view suddenly moves down:

https://github.com/user-attachments/assets/317193c3-c366-4862-88ec-70c9ce056012

I tried adding `.ignoresSafeArea()` to `PointOfSaleLoadingView` which seems to do the trick and immediately put the loading view in a position that ignores the existence of tab bar.

iamgabrielma replied:
That's neat, it also works on my end and the little jump while the LoadingView is in-flight disappears 💯 , updated on https://github.com/woocommerce/woocommerce-ios/pull/13905/commits/67ef3461324ed05d33a7d923e014f82f48954178

staskus reviewed (commented):
I added one suggestion to avoid a sudden drop of loading view. Please let me know if it works for you.

iamgabrielma committed:
Fix view jump on loading

Since we hide the app’s TabBar with an animation after a completion handler is called, the LoadingView will “jump” mid-loading to adapt to the new safe areas. By ignoring these we fix the visual glitch.

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

staskus reviewed (approved):
🚀
